plugin/src/main/java/com/denizenscript/denizen/objects/NPCTag.java

```diff
@@ -1277,6 +1277,68 @@ else if (attribute.startsWith("list", 2)) {
             return result;
         });
 
+        // <--[tag]
+        // @attribute <NPCTag.waypoint_provider>
+        // @returns ElementTag
+        // @description
+        // Returns the NPC's current Waypoint Provider type (default types: linear, wander, guided.)
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "waypoint_provider", (attribute, object) -> {
+            Waypoints wp = object.getCitizen().getTraitNullable(Waypoints.class);
+            if (wp != null) {
+                return new ElementTag(wp.getCurrentProviderName());
+            }
+            return null;
+        });
+
+        // <--[tag]
+        // @attribute <DurationTag.wander_delay>
+        // @returns DurationTag
+        // @description
+        // Returns the delay for the NPC's wander Waypoint Provider.
+        // -->
+        tagProcessor.registerTag(DurationTag.class, "wander_delay", (attribute, object) -> {
+            Waypoints wp = object.getCitizen().getTraitNullable(Waypoints.class);
+            if (wp != null) {
+                if (wp.getCurrentProvider() instanceof WanderWaypointProvider wanderWaypointProvider) {
+                    return new DurationTag(wanderWaypointProvider.getDelay());
```

Comment:
If I'm not mistaken `wanderWaypointProvider.getDelay()` is in ticks - so you'll have to use the DurationTag ticks constructor (cast to `(long)` is the easiest way)